internal/pkg/machine/controllers/etcd.go

```diff
@@ -98,7 +98,7 @@ outer:
 				continue
 			}
 
-			if config == nil || !config.Provider().Machine().Type().IsControlPlane() {
+			if config == nil || config.Provider().Machine() == nil || !config.Provider().Machine().Type().IsControlPlane() {
```

Comment:
wouldn;t this panic if `config` is nil :thinking:

Reply:
I dont think it would, as it would short-circuit, but I inverted and improved the logic here to be more readable.